## 02 Randomness Exercise 1

<h3 style="color: #001a79;">Exercise 1</h3>

<hr style="border-top: 1px solid #001a79;" />

<i h3 style="color: #001a79;">It is somewhat interesting that `(5 * 4 * 3 * 2 * 1)` perfectly divides `(10 * 9 * 8 * 7 * 6)` - there's no remainder.

<i h3 style="color: #001a79;">If we only wanted exactly four heads as opposed to five, the equivalent calculation would be `(10 * 9 * 8 * 7) / (4 * 3 * 2 * 1)`.

<i h3 style="color: #001a79;">a) Does that evenly divide too? 

<i h3 style="color: #001a79;">b) What is the formula in general?

<i h3 style="color: #001a79;">c) Does it always come out as a positive whole number? </i>

<hr style="border-top: 1px solid #001a79;" />

a) This will be evenly divided.

In [2]:
(10 * 9 * 8 * 7) // (4 * 3 * 2 * 1)

210

b) The formula in general is called Binomial coefficient which is used to calculate the number of ways to choose a subset of items from a set.

The general formual can be expressed as 

n! / (k! * (n-k)!)

For example if we want to calculate how many ways we can get 4 heads from 10 coin flips, the binomial coefficient for this would be 10 choose 4, which can be written as the following formula:

(10! / (4! * (10-4)!)) = 
(1 * 2 * 3 * 4 * 5 * 6 * 7 * 8 * 9 * 10) / (1 * 2 * 3 * 4) * (1 * 2 * 3 * 4 * 5 * 6)  = 
(7 * 8 * 9 * 10) / (1 * 2 * 3 * 4) = 210

C) As the Binomial coefficient will give us how many ways we can choose a subset of items from a larger set, this number will always be a positive whole number.